### ISSUE TITLE
RDDbConfig cannot update database version

### ISSUE THREAD

**Issue by WMTH** [ElvishArtisan/rivendell#414]
In version 3.0.0rc02 RDDbConfig cannot update to the current database version.  When opening the application, the indication is the version requires an upgrade.  After answering "Yes" to the prompt,   a response is returned indicating the database has been upgraded.  After clicking "OK" to the success response, the user is again prompted to upgrade the database.  This sequence will continue without the database actually being upgraded.  Database upgrade is successful when using rddbmgr --modify.  Schema version test 307 -> 308.

**Comment by deltecent** [comment]
WMTH, we have not been able to reproduce this problem. After upgrading the database using ```rddbmgr```, does ```rddbconfig``` show your database version as 308?

**Comment by WMTH** [comment]
Yes, the version is indicated to be 308.  Tested on several installations and all operate in the same manner.  Compiled on the most current build and patch level of CentOS 7.  I can reproduce by rolling back to an earlier schema version or restoring from previous database version and running RDDBconfig as outlined in the original report.

**Comment by deltecent** [comment]
When you roll back to an earlier version using ```rddbmgr```, which version do you roll back to in order to reproduce this problem. Would it possible to get a copy of a database that you restore to?

**Comment by deltecent** [comment]
I think you're going to have attach your databases to this issue at GitHub. They didn't come across through email.

**Comment by WMTH** [comment]
Here are the DBs.

[Test-DBs.zip](https://github.com/ElvishArtisan/rivendell/files/3164324/Test-DBs.zip)

**Comment by WMTH** [comment]
I was afraid of that.  Archive posted on GitHub.

**Comment by deltecent** [comment]
I am not able to reproduce this problem with your database files. I can load your 307 database and upgrade it to 308 without any issues. Since ```rddbconfig``` calls ```rddbmgr```, there shouldn't be a difference. I think at this point in time we're going to have to do a remote VNC session into your system there.

**Comment by WMTH** [comment]
Sure, I am available today (Friday) from 0630 CDT until 1100 CDT and after 1500 CDT.  If that works, please let me know.  If not, we can look at another day.

**Comment by WMTH** [comment]
Additional test results:  Restored database to schema 307, ran rddbconfig from the desktop as the rivendell user and the update to schema 308 fails however the log output claims the update was successful but as further indicated, it was not and rddbconfig also indicates the schema is 307.  Log output:
May 12 08:28:11 audiodesk rddbconfig-root: Restored Rivendell database from /home/rd/backup/rivendell-3.0.0rc0b9.sql
May 12 08:28:18 audiodesk rddbconfig-root: Modified database to version 308
May 12 08:28:21 audiodesk systemd: Started Rivendell Radio Automation System.
May 12 08:28:21 audiodesk rdservice: unable to open database [Database version mismatch, should be 308, is 307]
May 12 08:28:21 audiodesk systemd: rivendell.service: main process exited, code=exited, status=2/INVALIDARGUMENT
May 12 08:28:21 audiodesk systemd: Unit rivendell.service entered failed state.

Ran another test again restoring to schema 307 BUT running rddbconfig as root (from a command line), the update is then successful:
May 12 08:40:15 audiodesk rddbconfig: Modified database to version 308
May 12 08:40:16 audiodesk systemd: rivendell.service holdoff time over, scheduling restart.
May 12 08:40:16 audiodesk systemd: Stopped Rivendell Radio Automation System.
May 12 08:40:16 audiodesk systemd: Started Rivendell Radio Automation System.

To recap the environment: Current version and patch level of CentOS 7, Mate Desktop.

**Comment by WMTH** [comment]
Performed some additional testing and found that creating a copy of rddbmgr in /usr/sbin addresses the problem.  This is where the binary resides on the appliance version (and it is called by rddbconfig) rather than /usr/local/sbin in the source-built version.

**Comment by deltecent** [comment]
Ah, so the bug is if ```rddbconfig``` cannot find ```rddbmgr```, there is no error. I didn't even think of that.

Also, I believe ```rddbconfig``` should be looking for ```rddbmgr``` using the prefix specified during "./configure". I'll look at that too.

Thank you for finding these.

I'll see about pushing a PR pushed today that addresses both problems.

**Comment by ElvishArtisan** [comment]
> Also, I believe rddbconfig should be looking for rddbmgr using the prefix specified during "./configure". I'll look at that too.

You can get this by including the ```rdpaths.h``` header and looking in ```RD_PREFIX```.

**Comment by deltecent** [comment]
I've confirmed that ```rddbmgr``` does not use ```RD_PREFIX```.

I've also confirmed that ```rddbconfig``` does not check for an error when spawning ```rddbmgr```.

Fixes on the way shortly.

**Comment by deltecent** [comment]
Fixed in PR #437
